apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -128,7 +128,18 @@ export class OAuthService {
 			const user: User = await this.userRepo.findByExternalIdOrFail(provisioningDto.externalUserId, systemId);
 			return user;
 		} catch (error) {
-			throw new OAuthSSOError(`Failed to find user with Id ${provisioningDto.externalUserId}`, 'sso_user_notfound');
+			let additionalInfo = '';
+			const decodedToken: JwtPayload | null = jwt.decode(idToken, { json: true });
+			const email = decodedToken?.email as string;
```

Comment:
technically this is string | undefined, only the guard next line makes sure its not undefined. I would actually add this information to the code for readability

Reply:
fixed